text/0000-view-change-method.md

```diff
@@ -0,0 +1,69 @@
+and change methods and how they should be used. In short, we would like to restrict view methods from accessing certain
+context variables and do not distinguish between view and change methods on the contract level. Developers have the option
+to differentiate between the two in frontend or through near-shell.
+
+# Motivation
+[motivation]: #motivation
+
+From the feedback we received it seems that developers are confused by the results they get from view calls, which are
+mainly caused by the context variables such as `signer_account_id`, `current_account_id`, `attached_deposit` do not make
+sense in a view call. To avoid such confusion and create better developer experience, it is better if those context variables
+are prohibited in view calls.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Among context variables, some do make sense in a view call, such as `block_index`, while the majority does not. Here we
+explicitly list the variables are not allowed in a view call and, in case they are invoked, the contract will panic with
+`Context variable <context_variable> is not allowed in view calls`.
+
+The only allowed context variables are:
+    - `block_index`
+    - `storage_usage`
+
+All others are prohibited.
+
+From the developer perspective, if they want to call view functions from command line on some contract, they would just
+call `near view <contractName> <methodName> [args]`. If they are building an app and want to call a view function from the
+frontend, they should follow the same pattern as we have right now, specifying `viewMethods` and `changeMethods` in
+`loadContract`.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+To implement this NEP, we need to change how context is handled in runtime. More specifically, we can add a field `is_view: bool`
+in `VMContext` to indicate whether we are processing a view call. If `is_view` is true, then all the access to the prohibited
+variables will error with the error message described above.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+In terms of not allowing context variables, I don't see any drawback as those variables do not have a proper meaning
+in view functions. For alternatives, see the section below.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This design is very simple and requires very little change to the existing infrastructure. An alternative solution is
+to distinguish between view methods and change methods on the contract level. One way to do it is through decorators, as
+described [here](https://github.com/nearprotocol/NEPs/pull/3). However, enforcing such distinction on the contract level
+requires much more work and is not currently feasible for Rust contracts. 
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Should we allow any context variables to be accessed at all? 
```

Comment:
We totally should :) `input`, `block_index`, `storage_usage`, `account_balance`, `random_seed` , etc are all useful.

The bigger question is, should we prohibit "Promise API" since calling it does not make sense in view method? @evgenykuzyakov  However, I think it is not as confusing as `signer_account_id` so we can keep it for now.

Reply:
Good point. I suggest we prohibit it because otherwise people might call it in view by mistake and getting strange errors.